core/src/main/scala/org/bitcoins/core/crypto/SchnorrDigitalSignature.scala

```diff
@@ -0,0 +1,59 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoins.core.util.{BitcoinSLogger, BitcoinSUtil, Factory}
+import scodec.bits.ByteVector
+
+sealed abstract class SchnorrDigitalSignature extends BitcoinSLogger {
```

Comment:
IMO there's no real advantage to using the sealed trait/factory object/impl case class pattern here, might as well define `SchnorrDigitialSignature` as a case class (which should lead to fewer allocations as well)